### DISCUSSION TITLE
Why the usage of Net::HTTP::Get.new instead of Faraday.get to download files?

### DISCUSSION THREAD

**Discussion by JuanKman94** [ignacio-chiazzo/ruby_whatsapp_sdk#66] [General]
Is there a specific reason why you used `Net::HTTP::Get.new` in https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/blob/abd53efce563e19349cef8785dea0c43f6e71a85/lib/whatsapp_sdk/api/client.rb#L43 instead of a regular `Faraday#get`?

I ask because I was writing my own client to WApp API (before finding your repo) and I've been stuck for a couple of days trying to figure out why I can't download media from facebook's [download media API](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/media#download-media).

So far I've been struggling to get the following to work:

```ruby
token = "abc123"

client = Faraday.new(url: url) do |faraday|
  faraday.request :multipart
  faraday.request :url_encoded
  faraday.request :authorization, "Bearer", token
  faraday.response :json
  faraday.options.timeout = DEFAULT_TIMEOUT
  faraday.options.open_timeout = DEFAULT_OPEN_TIMEOUT
end

media_id = "abc123"
media_resp = client.get("/v15.0/#{media_id}")

url = URI.parse(media_resp.body["url"])
response = client.get(url)
response.status # 404
```

But if I use your code it works like a charm :thinking: I'm not sure if Faraday is not clearing an internal state or something

**Comment by ignacio-chiazzo** [comment]
Hola, Sorry, I just saw this post. If you haven't found a solution, like. 

I'm using the library in prod, and I haven't had any issues with Net/Faraday.
